zerver/actions/message_send.py

```diff
@@ -1532,8 +1532,12 @@ def check_private_message_policy(
     realm: Realm, sender: UserProfile, user_profiles: Sequence[UserProfile]
 ) -> None:
     if realm.private_message_policy == Realm.PRIVATE_MESSAGE_POLICY_DISABLED:
-        if sender.is_bot or (len(user_profiles) == 1 and user_profiles[0].is_bot):
-            # We allow direct messages only between users and bots,
+        if (
+            sender.is_bot
+            or (len(user_profiles) == 1 and user_profiles[0].is_bot)
+            or sender == user_profiles[0]
```

Comment:
This check looks wrong -- it looks like this would allow violating the policy as long as you are one of the recipients. You need the `len(user_profiles) == 1` in this last line.

You also need a backend tend for any server change.

Reply:
Sure, I will look into it.

Reply:
Thank you for catching that oversight. I've addressed the issue and pushed the fix.

Additionally, I took a closer look at the [architecture docs](https://zulip.readthedocs.io/en/latest/overview/architecture-overview.html#supervisor) and [sending messages](https://zulip.readthedocs.io/en/latest/subsystems/sending-messages.html) to understand the code flow better, but I couldn't figure out where to make the change exactly. Can you guide me on where to focus?